flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerImpl.java

```diff
@@ -807,23 +811,32 @@ private boolean isFulfillableByRegisteredSlots(ResourceProfile resourceProfile)
 		return false;
 	}
 
-	private Optional<PendingTaskManagerSlot> allocateResource(ResourceProfile resourceProfile) {
-		final Collection<ResourceProfile> requestedSlots = resourceActions.allocateResource(resourceProfile);
+	private Optional<PendingTaskManagerSlot> allocateResource(ResourceProfile requestedSlotResourceProfile) {
+		if (defaultWorkerResourceSpec == null) {
+			// standalone mode, cannot allocate resource
+			return Optional.empty();
+		}
```

Comment:
I think here we are leaking details from the enclosing component into the `SlotManager`. I think the `SlotManager` should not decide for the `ResourceManager` whether it can allocate new resources or not.